### ISSUE TITLE
 Could not find a version that satisfies the requirement ortools==7.4.7247 (from pm4py==1.2.10) 

### ISSUE THREAD

**Issue by fmannhardt** [process-intelligence-solutions/pm4py#140]
Installing PM4Py from Scratch on a Linux machine (Travis CI) leads to the following error:
```
$ pip install $PM4PY_PIP
Collecting pm4py==1.2.10
  Downloading pm4py-1.2.10-py3-none-any.whl (383 kB)
     |████████████████████████████████| 383 kB 2.7 MB/s 
Collecting pulp
  Using cached PuLP-2.0-py3-none-any.whl (39.2 MB)
Collecting networkx
  Using cached networkx-2.4-py3-none-any.whl (1.6 MB)
ERROR: Could not find a version that satisfies the requirement ortools==7.4.7247 (from pm4py==1.2.10) (from versions: 7.5.7466)
ERROR: No matching distribution found for ortools==7.4.7247 (from pm4py==1.2.10)
The command "pip install $PM4PY_PIP" failed and exited with 1 during .
```

You can see the full log here:
https://travis-ci.org/bupaverse/pm4py/jobs/644593765

Numpy is the only package installed beforehand.

**Comment by fit-alessandro-berti** [comment]
Hi, latest release of Ortools is flawed so we specified that one in reqs

Anyhow the Linux package is there
https://pypi.org/project/ortools/7.4.7247/#files

Could you try pip install --no-cache-dir -U pm4py  ?

**Comment by fit-alessandro-berti** [comment]
Just checked on my Linux machine. Everything is regular (x86-64 platform), possibly you are using another platform or you simply need to use --no-cache-dir? For any other platform, you could check the ARM32/ARM64 installation procedure on the site that covers the installation of PM4Py when some package is not available
 
Successfully installed pm4py-1.2.10
root@northwood:/home/javert# pip3 uninstall ortools pm4py
Uninstalling ortools-7.4.7247:
  Would remove:
    /usr/local/lib/python3.7/dist-packages/ortools-7.4.7247.dist-info/*
    /usr/local/lib/python3.7/dist-packages/ortools/*
Proceed (y/n)? y
  Successfully uninstalled ortools-7.4.7247
Uninstalling pm4py-1.2.10:
  Would remove:
    /usr/local/lib/python3.7/dist-packages/pm4py-1.2.10.dist-info/*
    /usr/local/lib/python3.7/dist-packages/pm4py/*
Proceed (y/n)? y       
  Successfully uninstalled pm4py-1.2.10
root@northwood:/home/javert# pip3 install --no-cache-dir -U pm4py
Collecting pm4py
  Downloading https://files.pythonhosted.org/packages/13/1e/517cc423711a715d0477b0c43e921431e17d7ec68f01abfa61d283acd2f4/pm4py-1.2.10-py3-none-any.whl (383kB)
    100% |████████████████████████████████| 389kB 209kB/s 
Requirement already satisfied, skipping upgrade: pyvis in /usr/local/lib/python3.7/dist-packages (from pm4py) (0.1.7.0)
Requirement already satisfied, skipping upgrade: scipy in /usr/lib/python3/dist-packages (from pm4py) (1.1.0)
Requirement already satisfied, skipping upgrade: pulp in /usr/local/lib/python3.7/dist-packages (from pm4py) (1.6.10)
Requirement already satisfied, skipping upgrade: networkx in /usr/local/lib/python3.7/dist-packages (from pm4py) (2.4)
Requirement already satisfied, skipping upgrade: pytz in /usr/local/lib/python3.7/dist-packages (from pm4py) (2019.3)
Requirement already satisfied, skipping upgrade: graphviz in /usr/local/lib/python3.7/dist-packages (from pm4py) (0.13)
Requirement already satisfied, skipping upgrade: pyarrow in /usr/local/lib/python3.7/dist-packages (from pm4py) (0.13.0)
Requirement already satisfied, skipping upgrade: lxml in /usr/local/lib/python3.7/dist-packages (from pm4py) (4.4.1)
Requirement already satisfied, skipping upgrade: numpy in /usr/local/lib/python3.7/dist-packages (from pm4py) (1.17.3)
Requirement already satisfied, skipping upgrade: scikit-learn in /usr/local/lib/python3.7/dist-packages (from pm4py) (0.21.3)
Requirement already satisfied, skipping upgrade: pandas in /usr/local/lib/python3.7/dist-packages (from pm4py) (0.24.2)
Requirement already satisfied, skipping upgrade: matplotlib in /usr/lib/python3/dist-packages (from pm4py) (3.0.2)
Requirement already satisfied, skipping upgrade: pydotplus in /usr/local/lib/python3.7/dist-packages (from pm4py) (2.0.2)
Requirement already satisfied, skipping upgrade: intervaltree in /usr/local/lib/python3.7/dist-packages (from pm4py) (3.0.2)
Collecting ortools==7.4.7247 (from pm4py)
  Downloading https://files.pythonhosted.org/packages/d8/d5/4015ba30ba3ddc4279aa36c284af1ed157ca7d9c926f8ccb5ac90da8a38e/ortools-7.4.7247-cp37-cp37m-manylinux1_x86_64.whl (27.7MB)
    17% |█████▌                          | 4.8MB 187kB/s eta 0:02:03^C
Operation cancelled by user

**Comment by fit-alessandro-berti** [comment]
Another reason is that you are using Python 3.8; in that case, the Ortools package of that version is not available for 3.8

On the other hand, ortools 7.5 offers the binary also for Python 3.8, but there are some issues in that version

**Comment by fit-alessandro-berti** [comment]
And thats the nice thing, I went to check your Travis configuration, and it seems that you are using the latest Miniconda

But the latest Miniconda is based on Python 3.7, not 3.8, so that should not be the problem

**Comment by fmannhardt** [comment]
Thanks for the quick reply, I found the issue. Travis was using Python 3.8 and from the website it was not so clear that Python 3,8 is not supported. I will change the CI script.

Btw, if you are interested I can create a PR for adding CI using Travis for PM4Py as well?

**Comment by fmannhardt** [comment]
Works with fixed Python 3.7:
https://travis-ci.org/bupaverse/pm4py/builds/644911356
